sdk/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -68,18 +69,50 @@
      */
     void startNewSession(@NonNull Date date, @Nullable User user, boolean autoCaptured) {
         synchronized (sessionStore) {
-            sessionStartMs = date.getTime();
+            currentSession = generateSession(date, user, autoCaptured);
+            trackSessionIfNeeded(autoCaptured, currentSession);
+        }
+    }
+
+    @NonNull
+    private Session generateSession(@NonNull Date date, @Nullable User user, boolean autoCaptured) {
+        sessionStartMs = date.getTime();
+        Session session = new Session(UUID.randomUUID().toString(), date, user);
+        session.setAutoCaptured(autoCaptured);
+        return session;
+    }
 
-            Session session = new Session(UUID.randomUUID().toString(), date, user);
-            session.setAutoCaptured(autoCaptured);
+    private void trackSessionIfNeeded(boolean autoCaptured, Session session) {
+        String releaseStage = getReleaseStage();
+        boolean notifyForRelease = client != null && configuration.shouldNotifyForReleaseStage(releaseStage);
```

Comment:
why does this check if the client is not null here? shouldn't you always put a session in the queue/store even if the client is not initialized yet?